core/slice.go

```diff
@@ -596,15 +596,17 @@ func (sl *Slice) updatePhCache(pendingHeaderWithTermini types.PendingHeader, inS
 	deepCopyPendingHeaderWithTermini.Header.SetTime(uint64(time.Now().Unix()))
 	if exist {
 		// If we are inslice we will only update the cache if the entropy is better
-		if !inSlice || newPhEntropy.Cmp(oldPh.Header.CalcPhS()) >= 0 {
+		if !inSlice && newPhEntropy.Cmp(oldPh.Header.CalcPhS()) >= 0 {
 			sl.writePhCache(pendingHeaderWithTermini.Termini[c_terminusIndex], deepCopyPendingHeaderWithTermini)
-			log.Debug("PhCache update:", "inSlice:", inSlice, "Ph Number:", deepCopyPendingHeaderWithTermini.Header.NumberArray(), "Termini:", pendingHeaderWithTermini.Termini[c_terminusIndex])
-			// if we are not inslice we are getting a coordinate update from dom which has already been decided by a coord
+			log.Info("PhCache update:", "!inSlice:", inSlice, "Ph Number:", deepCopyPendingHeaderWithTermini.Header.NumberArray(), "Termini:", deepCopyPendingHeaderWithTermini.Termini[c_terminusIndex])
+		} else if inSlice && pendingHeaderWithTermini.Header.ParentEntropy().Cmp(oldPh.Header.ParentEntropy()) > 0 {
+			sl.writePhCache(pendingHeaderWithTermini.Termini[c_terminusIndex], deepCopyPendingHeaderWithTermini)
```

Comment:
It looks like we do the exact same write in the `if` and `else if' blocks? Please collapse the cases:
```
if !inSlice && newPhEntropy.Cmp(oldPh.Header.CalcPhS()) >= 0 ||
    inSlice && pendingHeaderWithTermini.Header.ParentEntropy().Cmp(oldPh.Header.ParentEntropy()) > 0
{
	sl.writePhCache(pendingHeaderWithTermini.Termini[c_terminusIndex], deepCopyPendingHeaderWithTermini)
}
log.Info("PhCache update:", "inSlice:", inSlice, "Ph Number:", deepCopyPendingHeaderWithTermini.Header.NumberArray(), "Termini:", deepCopyPendingHeaderWithTermini.Termini[c_terminusIndex])
```

Also your logs are a bit confused, I think. In one case you print '!inSlice', but give the 'inSlice' value (logical invert). I think you just want the one log print like I have above.

Reply:
Good catch, wrong logs and redundant logic, changed it